src/arvnetwork.c

```diff
@@ -39,6 +39,28 @@ struct _ArvNetworkInterface{
 };
 
 #ifdef G_OS_WIN32
+// mandatory declaration
+__attribute__((constructor)) void arv_initialize_networking(void);
```

Comment:
Is it really mandatory ? I've never added declaration when I've used constructors.

Reply:
Yes, it is (also stated in [WSAStartup docs](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsastartup)) . It was the reason for the funny characters as `getnameinfo` was failing (saying the `WSAStartup` was not called yet). This only manifested in real Windows 10 installation, not in wine.

Reply:
I get this warning without the declaration:
```
../src/arvnetwork.c:47:6: warning: no previous prototype for ‘arv_initialize_networking’ [-Wmissing-prototypes]
   47 | void arv_initialize_networking(void){
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/arvnetwork.c:80:6: warning: no previous prototype for ‘arv_cleanup_networking’ [-Wmissing-prototypes]
   80 | void arv_cleanup_networking(void){
      |      ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I've just remembered constructor and destructor should be made static.

Reply:
Fixed.